stripe/src/main/java/com/stripe/android/paymentsheet/PaymentSheet.kt

```diff
@@ -26,28 +27,33 @@ class PaymentSheet internal constructor(
     )
 
     /**
-     * Create PaymentSheet with a [Configuration].
+     * Create PaymentSheet to process a [PaymentIntent].
+     * If the [PaymentIntent] is already confirmed, [PaymentSheetResultCallback] will be invoked
+     * with [PaymentSheetResult.Completed].
      *
-     * If [paymentIntentClientSecret] represents a [PaymentIntent] that is already confirmed,
-     * [PaymentSheetResultCallback] will be invoked with [PaymentResult.Completed].
+     * @param[paymentIntentClientSecret] the client secret for the [PaymentIntent]
+     * @param[configuration] optional [PaymentSheet] settings
      */
-    internal fun present(
+    internal fun presentWithPaymentIntent(
```

Comment:
I'll let Yuki answer, but I'm assuming this will be part of the docs update he's working on.

Reply:
Is this API is for MC Complete? This is the iOS API:

```
    /// Initializes a PaymentSheet
    /// - Note Intentionally non-public; for access, see https://stripe.com/docs/mobile/payments-ui-beta
    /// - Parameter intentClientSecret: The [client secret](https://stripe.com/docs/api/payment_intents/object#payment_intent_object-client_secret) of a Stripe PaymentIntent object
    /// - Note: This can be used to complete a payment - don't log it, store it, or expose it to anyone other than the customer.
    /// - Note: You may also pass the [client secret](https://stripe.com/docs/api/setup_intents/object#setup_intent_object-client_secret) of a Stripe SetupIntent object.
    /// - Parameter configuration: Configuration for the PaymentSheet. e.g. your business name, Customer details, etc.
    /* public */ required init(intentClientSecret: String, configuration: Configuration) {
```

```
    /// Presents a sheet for a customer to complete their payment
    /// - Parameter presentingViewController: The view controller to present a payment sheet
    /// - Parameter completion: Called with the result of the payment after the payment sheet is dismissed
    @available(iOSApplicationExtension, unavailable)
    @available(macCatalystApplicationExtension, unavailable)
    public func present(
        from presentingViewController: UIViewController,
        completion: @escaping (PaymentSheetResult) -> ()
    ) {
```

Reply:
Yes, for MC Complete. It's inverted in that you init with the presenter and the callback, then call present with the configuration and secret. @mshafrir-stripe is there a reason for that? Should we invert to match iOS?

Reply:
The way we do it on Android is necessary to support the Activity Result APIs